lib/teleterm/services/clientcache/clientcache.go

```diff
@@ -163,31 +163,31 @@ func (c *Cache) Clear() error {
 	return trace.NewAggregate(errors...)
 }
 
-func (c *Cache) addToCache(clusterURI uri.ResourceURI, proxyClient *client.ProxyClient) {
+func (c *Cache) addToCache(clusterURI uri.ResourceURI, clusterClient *client.ClusterClient) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	c.clients[clusterURI] = proxyClient
+	c.clients[clusterURI] = clusterClient
 
 	// This goroutine removes the connection from the cache when
 	// it is unexpectedly interrupted (for example, by the remote site).
 	// It will also react to client.Close() called from our side, but it will be noop.
-	go func() {
-		err := proxyClient.Client.Wait()
-		c.mu.Lock()
-		defer c.mu.Unlock()
-
-		if c.clients[clusterURI] != proxyClient {
-			return
-		}
-
-		delete(c.clients, clusterURI)
-		c.cfg.Log.WithField("cluster", clusterURI.String()).WithError(err).
-			Info("Connection has been closed, removed client from cache.")
-	}()
+	// go func() {
+	// 	err := proxyClient.Client.Wait()
+	// 	c.mu.Lock()
+	// 	defer c.mu.Unlock()
+
+	// 	if c.clients[clusterURI] != proxyClient {
+	// 		return
+	// 	}
+
+	// 	delete(c.clients, clusterURI)
+	// 	c.cfg.Log.WithField("cluster", clusterURI.String()).WithError(err).
+	// 		Info("Connection has been closed, removed client from cache.")
+	// }()
```

Comment:
I needed this because when the connection got closed (for example, by stopping the Teleport cluster) I wasn't able to use it again (it was stuck in a "broken" state).  

From what I remember, you were saying that this mechanism is not needed for gRPC connections because it would reconnect automatically. I tested this now and it seems to work, after I stopped the cluster and run it again, the connection was resumed.
The only issue I see, is that it didn't happen immediately, for around a minute I was getting `connection error: desc = "transport: Error while dialing: failed to dial: unable to establish proxy stream\n\trpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp 127.0.0.1:3023: connect: connection refused\""` errors when I tried to send any RPC and after that it started working. I believe this is related to keep alive pings (we have a similar issue https://github.com/gravitational/teleport/issues/39009).

Reply:
nit: maybe we can add a comment on this somewhere in the cacheclient